arcgis-ios-sdk-samples/Utility network/Perform valve isolation trace/PerformValveIsolationTraceViewController.swift

```diff
@@ -18,140 +18,228 @@ import ArcGIS
-    @IBOutlet weak var mapView: AGSMapView! {
+    @IBOutlet var mapView: AGSMapView! {
         didSet {
             mapView.map = makeMap()
+            mapView.graphicsOverlays.add(parametersOverlay)
         }
     }
     
-    // MARK: Instance properties
+    /// The button to start or reset the trace.
+    @IBOutlet var traceResetBarButtonItem: UIBarButtonItem! {
+        didSet {
+            traceResetBarButtonItem.possibleTitles = ["Trace", "Reset"]
+        }
+    }
+    /// The button to choose a utility category for filter barriers.
+    @IBOutlet var categoryBarButtonItem: UIBarButtonItem!
+    /// The switch to control whether to include isolated features in the
+    /// trace results when used in conjunction with an isolation trace.
+    @IBOutlet var isolationSwitch: UISwitch!
+    /// The label to display trace status.
+    @IBOutlet var statusLabel: UILabel!
+    
+    // MARK: Constant
     
     /// The URL to the feature service for running the isolation trace.
-    static let featureServiceURL = URL(string: "https://sampleserver7.arcgisonline.com/server/rest/services/UtilityNetwork/NapervilleGas/FeatureServer")!
+    static let featureServiceURL = URL(
+        string: "https://sampleserver7.arcgisonline.com/server/rest/services/UtilityNetwork/NapervilleGas/FeatureServer"
+    )!
     
-    // Constants for creating the default trace configuration.
-    let domainNetworkName = "Pipeline"
-    let tierName = "Pipe Distribution System"
+    static let filterBarrierIdentifier = "filter barrier"
     
-    // Constants for creating the default starting location.
-    let networkSourceName = "Gas Device"
-    let assetGroupName = "Meter"
-    let assetTypeName = "Customer"
-    let globalId = UUID(uuidString: "98A06E95-70BE-43E7-91B7-E34C9D3CB9FF")!
+    // MARK: Instance properties
     
-    /// An array to keep track of all available utility categories under current network definition.
-    var filterBarrierCategories: [AGSUtilityCategory] = []
-    var traceConfiguration: AGSUtilityTraceConfiguration?
-    var selectedCategory: AGSUtilityCategory?
-    var startingLocationElement: AGSUtilityElement!
     let utilityNetwork = AGSUtilityNetwork(url: featureServiceURL)
     let serviceGeodatabase = AGSServiceGeodatabase(url: featureServiceURL)
-    var layers = [AGSFeatureLayer]()
+    var traceCompleted = false
+    var identifyAction: AGSCancelable?
+    
+    /// The base trace parameters.
+    let traceParameters = AGSUtilityTraceParameters(traceType: .isolation, startingLocations: [])
+    /// The utility category selected for running the trace.
+    var selectedCategory: AGSUtilityCategory?
+    /// An array of available utility categories with current network definition.
+    var filterBarrierCategories: [AGSUtilityCategory]!
+    /// An array to hold the gas line and gas device feature layers created from
+    /// the service geodatabase.
+    var layers: [AGSFeatureLayer]!
```

Comment:
These don't need to be optional.

```suggestion
    /// An array of available utility categories with current network definition.
    var filterBarrierCategories: [AGSUtilityCategory] = []
    /// An array to hold the gas line and gas device feature layers created from
    /// the service geodatabase.
    var layers: [AGSFeatureLayer] = []
```